src/py/flwr/server/run_serverapp.py

```diff
@@ -136,32 +137,50 @@ def run_server_app() -> None:
             cert_path,
         )
 
-    log(
-        DEBUG,
-        "Flower will load ServerApp `%s`",
-        getattr(args, "server-app"),
+    server_app_attr: Optional[str] = getattr(args, "server-app")
+    if not (server_app_attr is None) ^ (args.run_id is None):
+        raise sys.exit(
+            "Please provide either a ServerApp reference or a Run ID, but not both. "
+            "For more details, use: ``flower-server-app -h``"
+        )
+
+    stub = GrpcDriverStub(
+        driver_service_address=args.superlink, root_certificates=root_certificates
     )
+    if args.run_id is not None:
+        # User provided `--run-id`, but not `server-app`
+        run_id = args.run_id
+    else:
+        # User provided `server-app`, but not `--run-id`
+        # Create run if run_id is not provided
+        stub.connect()
+        req = CreateRunRequest(fab_id=args.fab_id, fab_version=args.fab_version)
+        res = stub.create_run(req)
+        run_id = res.run_id
+
+    # Initialize GrpcDriver
+    driver = GrpcDriver(run_id=run_id, stub=stub)
+
+    # Dynamically obtain ServerApp path based on run_id
+    if args.run_id is not None:
+	# User provided `--run-id`, but not `server-app`
+        flwr_dir = get_flwr_dir(args.flwr_dir)
+        run_ = driver.run
```

Comment:
Can this be just `run`?
```suggestion
        run = driver.run
```